### DISCUSSION TITLE
Change SysEx formula from Lua

### DISCUSSION THREAD

**Discussion by inteyes** [RomanKubiak/ctrlr#335] [Q&A]
Hey guys.

Is anybody have an idea on how to properly change modulator sysex formula from Lua?

I have tried the most obvious approach like:

`panel:getModulatorByName("knobOsc2Semitone"):getMidiMessage():setPropertyString("midiMessageSysExFormula", "F0 42 3y 58 41 4F 00 LS MS F7")`

It works pretty well - modulator become to send values using new formula. But the panel cannot recognize and display incoming values from synthesizer using this new formula until I enter and exit "panel mode" (even though I make no changes there).

Is there any chance to get it to work? Really want to avoid doubling the whole knob collection on the panel to send slightly different data.

Thanks!

**Comment by dnaldoog** [comment]
Hi Inteyes!

I don't suppose firing `panel:getCanvas():repaint()` would work, but how about trying:
`panel:getModulatorByName("knobOsc2Semitone"):getMidiMessage():setProperty("midiMessageSysExFormula", "F0 42 3y 58 41 4F 00 LS MS F7",true)` -- but I have no idea actually.

**Reply by inteyes** [reply]
Hi, John!

Unfortanutaly it made no difference.

But I completely forgot about the internal "_what_" lua method. I managed to find a function that did the trick - `panel:getInputComparator():rehashComparator()`

I inserted it at the end of the sysex formula replacement function and now it recognizes all incoming messages fine. I don't know if it's correct way to do it, but it works to me :)

Thanks for your answer!

**Reply by dnaldoog** [reply]
That’s very interesting!

I have no idea about that method.

Well, I didn’t really think my suggestion was going to be very helpful, but that is a very interesting finding you made there.

:)
